packages/gatsby-dev-cli/src/local-npm-registry/install-packages.js

```diff
@@ -29,7 +29,49 @@ const installPackages = async ({
       { stdio: `pipe` },
     ])
 
-    const workspacesLayout = JSON.parse(JSON.parse(stdout).data)
+    let workspacesLayout
+    try {
+      workspacesLayout = JSON.parse(JSON.parse(stdout).data)
+    } catch (e) {
+      /*
+      Yarn 1.22 doesn't output pure json - it has leading and trailing text:
+      ```
+      $ yarn workspaces info --json
+      yarn workspaces v1.22.0
+      {
+        "z": {
+          "location": "z",
+          "workspaceDependencies": [],
+          "mismatchedWorkspaceDependencies": []
+        },
+        "y": {
+          "location": "y",
+          "workspaceDependencies": [],
+          "mismatchedWorkspaceDependencies": []
+        }
+      }
+      Done in 0.48s.
+      ```
+      So we need to do some sanitization. We find JSON by matching substring
+      that starts with `{` and ends with `}`
+    */
+
+      const regex = /^[^{]*({.*})[^}]*$/gs
+      const sanitizedStdOut = regex.exec(stdout)
+      if (sanitizedStdOut?.length >= 2) {
+        // pick content of first (and only) capturing group
+        const jsonString = sanitizedStdOut[1]
+        workspacesLayout = JSON.parse(jsonString)
```

Comment:
Should probably try/catch this too, emit a sensible error (for ourselves).

Reply:
Added some try/catching and just log some details specific to this code path. 

We will exit with generic error message below